app/Http/Controllers/Auth/LoginController.php

```diff
@@ -36,161 +28,78 @@ class LoginController extends BaseController
      */
     public function __construct()
     {
-        $this->middleware('guest')->except('showLoginForm', 'loginSecondary', 'logout');
+        $this->provider = new VatsimOAuthController();
     }
 
-    public function getLogin()
+    public function login(Request $request)
     {
-        if (Auth::guard('vatsim-sso')->check() && ! Auth::check()) {
-            return $this->attemptSecondaryAuth();
-        } else {
-            return redirect()->route('dashboard');
-        }
-    }
+        if (! $request->has('code') || ! $request->has('state')) {
+            $authorizationUrl = $this->provider->getAuthorizationUrl();
+            $request->session()->put('vatsimauthstate', $this->provider->getState());
```

Comment:
Worth putting `vatsimauthstate` into either a controller class constant or the config?

Reply:
I think this is fine as is, although the two suggestions would also be fine